### PR TITLE
[SYCL] Move pretty-printers from the GDB repository

### PR DESCRIPTION
**Pull request by mihails-strasuns-intel**
Similar to xmethods, pretty-printing scripts should be distributed
together with a matching runtmime library to ensure it stays in sync.

Directory names have been adjusted to make them more generic.

### CODE REVIEW

File: sycl/gdb/libsycl.so-gdb.py [hunk 1]
```diff
@@ -134,3 +139,86 @@ def match(self, class_type, method_name):
 
 
 gdb.xmethod.register_xmethod_matcher(None, AccessorOpIndexMatcher(), replace=True)
+
+### Pretty-printer implementations ###
+
+"""
+Print a cl::sycl::array
```
Comment (keryell) [inline sycl/gdb/libsycl.so-gdb.py:146]: What is a `cl::sycl::array`?

File: sycl/gdb/libsycl.so-gdb.py [hunk 2]
```diff
@@ -134,3 +139,86 @@ def match(self, class_type, method_name):
...
+        match = self.typeregex.match(self.type.tag)
+        if not match:
+            return "<error parsing type>"
+        return ('%s<%s, %s> = {impl=%s}'
+                % (match.group(1), self.elt_type, self.dimensions,
+                   self.value['impl'].address))
+
+sycl_printer = gdb.printing.RegexpCollectionPrettyPrinter("SYCL")
+sycl_printer.add_printer("cl::sycl::id",     '^cl::sycl::id<.*$',     SyclArrayPrinter)
+sycl_printer.add_printer("cl::sycl::range",  '^cl::sycl::range<.*$',  SyclArrayPrinter)
```
Comment (keryell) [inline sycl/gdb/libsycl.so-gdb.py:221]: OK, probably you need a concept above in my first comment, not an explicit type.
Something like "Print SYCL objects based on small array underlying type (id, range...)"?

File: sycl/gdb/libsycl.so-gdb.py (hunk 1 above)
Comment (mihails-strasuns-intel) [reply sycl/gdb/libsycl.so-gdb.py:146]: It is defined in `sycl/detail/array.hpp` (see the test case) and used as a base class for several other entities.

File: sycl/gdb/libsycl.so-gdb.py (hunk 2 above)
Comment (mihails-strasuns-intel) [reply sycl/gdb/libsycl.so-gdb.py:221]: "Print anything inheriting from array"?

File: sycl/gdb/libsycl.so-gdb.py (hunk 1 above)
Comment (bader) [inline sycl/gdb/libsycl.so-gdb.py:146]:
```suggestion
Print a cl::sycl::detail::array
```

File: sycl/gdb/libsycl.so-gdb.py (hunk 2 above)
Comment (keryell) [reply sycl/gdb/libsycl.so-gdb.py:221]: Be more precise here. It is not the obvious `std::array` for example...

File: sycl/gdb/libsycl.so-gdb.py (hunk 1 above)
Comment (keryell) [reply sycl/gdb/libsycl.so-gdb.py:146]: Except that it is not. As @bader suggested there is a `namespace detail {` there...

Comment (mihails-strasuns-intel) [reply sycl/gdb/libsycl.so-gdb.py]: Updated also with "inheriting from" comment.

**Pull request merged**